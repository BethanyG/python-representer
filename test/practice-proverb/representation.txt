Module([FunctionDef('placeholder_0', arguments([], [], arg('placeholder_1', lineno=1, col_offset=13, end_lineno=1, end_col_offset=24), [arg('placeholder_2', lineno=1, col_offset=26, end_lineno=1, end_col_offset=35)], [None], defaults=[]), [If(UnaryOp(Not(), Name('placeholder_1', Load(), lineno=3, col_offset=11, end_lineno=3, end_col_offset=22), lineno=3, col_offset=7, end_lineno=3, end_col_offset=22), [Return(List([], Load(), lineno=4, col_offset=15, end_lineno=4, end_col_offset=17), lineno=4, col_offset=8, end_lineno=4, end_col_offset=17)], [], lineno=3, col_offset=4, end_lineno=4, end_col_offset=17), Assign([Name('placeholder_3', Store(), lineno=5, col_offset=4, end_lineno=5, end_col_offset=11)], ListComp(JoinedStr([Constant('For want of a ', lineno=5, col_offset=15, end_lineno=5, end_col_offset=69), FormattedValue(Name('placeholder_4', Load(), lineno=5, col_offset=32, end_lineno=5, end_col_offset=41), -1, lineno=5, col_offset=15, end_lineno=5, end_col_offset=69), Constant(' the ', lineno=5, col_offset=15, end_lineno=5, end_col_offset=69), FormattedValue(Name('placeholder_5', Load(), lineno=5, col_offset=48, end_lineno=5, end_col_offset=57), -1, lineno=5, col_offset=15, end_lineno=5, end_col_offset=69), Constant(' was lost.', lineno=5, col_offset=15, end_lineno=5, end_col_offset=69)], lineno=5, col_offset=15, end_lineno=5, end_col_offset=69), [comprehension(Tuple([Name('placeholder_4', Store(), lineno=6, col_offset=19, end_lineno=6, end_col_offset=28), Name('placeholder_5', Store(), lineno=6, col_offset=30, end_lineno=6, end_col_offset=39)], Store(), lineno=6, col_offset=19, end_lineno=6, end_col_offset=39), Call(Name('zip', Load(), lineno=6, col_offset=43, end_lineno=6, end_col_offset=46), [Name('placeholder_1', Load(), lineno=6, col_offset=47, end_lineno=6, end_col_offset=58), Subscript(Name('placeholder_1', Load(), lineno=6, col_offset=60, end_lineno=6, end_col_offset=71), Slice(Constant(1, lineno=6, col_offset=72, end_lineno=6, end_col_offset=73), lineno=6, col_offset=72, end_lineno=6, end_col_offset=74), Load(), lineno=6, col_offset=60, end_lineno=6, end_col_offset=75)], [], lineno=6, col_offset=43, end_lineno=6, end_col_offset=76), [], 0)], lineno=5, col_offset=14, end_lineno=6, end_col_offset=77), lineno=5, col_offset=4, end_lineno=6, end_col_offset=77), If(Name('placeholder_2', Load(), lineno=7, col_offset=7, end_lineno=7, end_col_offset=16), [Expr(Call(Attribute(Name('placeholder_3', Load(), lineno=8, col_offset=8, end_lineno=8, end_col_offset=15), 'append', Load(), lineno=8, col_offset=8, end_lineno=8, end_col_offset=22), [JoinedStr([Constant('And all for the want of a ', lineno=8, col_offset=23, end_lineno=8, end_col_offset=81), FormattedValue(Name('placeholder_2', Load(), lineno=8, col_offset=52, end_lineno=8, end_col_offset=61), -1, lineno=8, col_offset=23, end_lineno=8, end_col_offset=81), Constant(' ', lineno=8, col_offset=23, end_lineno=8, end_col_offset=81), FormattedValue(Subscript(Name('placeholder_1', Load(), lineno=8, col_offset=64, end_lineno=8, end_col_offset=75), Constant(0, lineno=8, col_offset=76, end_lineno=8, end_col_offset=77), Load(), lineno=8, col_offset=64, end_lineno=8, end_col_offset=78), -1, lineno=8, col_offset=23, end_lineno=8, end_col_offset=81), Constant('.', lineno=8, col_offset=23, end_lineno=8, end_col_offset=81)], lineno=8, col_offset=23, end_lineno=8, end_col_offset=81)], [], lineno=8, col_offset=8, end_lineno=8, end_col_offset=82), lineno=8, col_offset=8, end_lineno=8, end_col_offset=82)], [Expr(Call(Attribute(Name('placeholder_3', Load(), lineno=10, col_offset=8, end_lineno=10, end_col_offset=15), 'append', Load(), lineno=10, col_offset=8, end_lineno=10, end_col_offset=22), [JoinedStr([Constant('And all for the want of a ', lineno=10, col_offset=23, end_lineno=10, end_col_offset=69), FormattedValue(Subscript(Name('placeholder_1', Load(), lineno=10, col_offset=52, end_lineno=10, end_col_offset=63), Constant(0, lineno=10, col_offset=64, end_lineno=10, end_col_offset=65), Load(), lineno=10, col_offset=52, end_lineno=10, end_col_offset=66), -1, lineno=10, col_offset=23, end_lineno=10, end_col_offset=69), Constant('.', lineno=10, col_offset=23, end_lineno=10, end_col_offset=69)], lineno=10, col_offset=23, end_lineno=10, end_col_offset=69)], [], lineno=10, col_offset=8, end_lineno=10, end_col_offset=70), lineno=10, col_offset=8, end_lineno=10, end_col_offset=70)], lineno=7, col_offset=4, end_lineno=10, end_col_offset=70), Return(Name('placeholder_3', Load(), lineno=11, col_offset=11, end_lineno=11, end_col_offset=18), lineno=11, col_offset=4, end_lineno=11, end_col_offset=18)], [], lineno=1, col_offset=0, end_lineno=11, end_col_offset=18)], [])