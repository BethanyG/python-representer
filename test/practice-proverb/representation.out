# BEGIN TREE BEFORE
Module(
  [
    FunctionDef(
      'proverb',
      arguments(
        [],
        [],
        arg(
          'rhyme_items',
          lineno=1,
          col_offset=13,
          end_lineno=1,
          end_col_offset=24),
        [
          arg(
            'qualifier',
            lineno=1,
            col_offset=26,
            end_lineno=1,
            end_col_offset=35)],
        [
          None],
        defaults=[]),
      [
        Expr(
          Call(
            Name(
              'print',
              Load(),
              lineno=2,
              col_offset=4,
              end_lineno=2,
              end_col_offset=9),
            [
              Name(
                'rhyme_items',
                Load(),
                lineno=2,
                col_offset=10,
                end_lineno=2,
                end_col_offset=21)],
            [],
            lineno=2,
            col_offset=4,
            end_lineno=2,
            end_col_offset=22),
          lineno=2,
          col_offset=4,
          end_lineno=2,
          end_col_offset=22),
        If(
          UnaryOp(
            Not(),
            Name(
              'rhyme_items',
              Load(),
              lineno=3,
              col_offset=11,
              end_lineno=3,
              end_col_offset=22),
            lineno=3,
            col_offset=7,
            end_lineno=3,
            end_col_offset=22),
          [
            Return(
              List(
                [],
                Load(),
                lineno=4,
                col_offset=15,
                end_lineno=4,
                end_col_offset=17),
              lineno=4,
              col_offset=8,
              end_lineno=4,
              end_col_offset=17)],
          [],
          lineno=3,
          col_offset=4,
          end_lineno=4,
          end_col_offset=17),
        Assign(
          [
            Name(
              'phrases',
              Store(),
              lineno=5,
              col_offset=4,
              end_lineno=5,
              end_col_offset=11)],
          ListComp(
            JoinedStr(
              [
                Constant(
                  'For want of a ',
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69),
                FormattedValue(
                  Name(
                    'element_1',
                    Load(),
                    lineno=5,
                    col_offset=32,
                    end_lineno=5,
                    end_col_offset=41),
                  -1,
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69),
                Constant(
                  ' the ',
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69),
                FormattedValue(
                  Name(
                    'element_2',
                    Load(),
                    lineno=5,
                    col_offset=48,
                    end_lineno=5,
                    end_col_offset=57),
                  -1,
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69),
                Constant(
                  ' was lost.',
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69)],
              lineno=5,
              col_offset=15,
              end_lineno=5,
              end_col_offset=69),
            [
              comprehension(
                Tuple(
                  [
                    Name(
                      'element_1',
                      Store(),
                      lineno=6,
                      col_offset=19,
                      end_lineno=6,
                      end_col_offset=28),
                    Name(
                      'element_2',
                      Store(),
                      lineno=6,
                      col_offset=30,
                      end_lineno=6,
                      end_col_offset=39)],
                  Store(),
                  lineno=6,
                  col_offset=19,
                  end_lineno=6,
                  end_col_offset=39),
                Call(
                  Name(
                    'zip',
                    Load(),
                    lineno=6,
                    col_offset=43,
                    end_lineno=6,
                    end_col_offset=46),
                  [
                    Name(
                      'rhyme_items',
                      Load(),
                      lineno=6,
                      col_offset=47,
                      end_lineno=6,
                      end_col_offset=58),
                    Subscript(
                      Name(
                        'rhyme_items',
                        Load(),
                        lineno=6,
                        col_offset=60,
                        end_lineno=6,
                        end_col_offset=71),
                      Slice(
                        Constant(
                          1,
                          lineno=6,
                          col_offset=72,
                          end_lineno=6,
                          end_col_offset=73),
                        lineno=6,
                        col_offset=72,
                        end_lineno=6,
                        end_col_offset=74),
                      Load(),
                      lineno=6,
                      col_offset=60,
                      end_lineno=6,
                      end_col_offset=75)],
                  [],
                  lineno=6,
                  col_offset=43,
                  end_lineno=6,
                  end_col_offset=76),
                [],
                0)],
            lineno=5,
            col_offset=14,
            end_lineno=6,
            end_col_offset=77),
          lineno=5,
          col_offset=4,
          end_lineno=6,
          end_col_offset=77),
        If(
          Name(
            'qualifier',
            Load(),
            lineno=7,
            col_offset=7,
            end_lineno=7,
            end_col_offset=16),
          [
            Expr(
              Call(
                Attribute(
                  Name(
                    'phrases',
                    Load(),
                    lineno=8,
                    col_offset=8,
                    end_lineno=8,
                    end_col_offset=15),
                  'append',
                  Load(),
                  lineno=8,
                  col_offset=8,
                  end_lineno=8,
                  end_col_offset=22),
                [
                  JoinedStr(
                    [
                      Constant(
                        'And all for the want of a ',
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81),
                      FormattedValue(
                        Name(
                          'qualifier',
                          Load(),
                          lineno=8,
                          col_offset=52,
                          end_lineno=8,
                          end_col_offset=61),
                        -1,
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81),
                      Constant(
                        ' ',
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81),
                      FormattedValue(
                        Subscript(
                          Name(
                            'rhyme_items',
                            Load(),
                            lineno=8,
                            col_offset=64,
                            end_lineno=8,
                            end_col_offset=75),
                          Constant(
                            0,
                            lineno=8,
                            col_offset=76,
                            end_lineno=8,
                            end_col_offset=77),
                          Load(),
                          lineno=8,
                          col_offset=64,
                          end_lineno=8,
                          end_col_offset=78),
                        -1,
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81),
                      Constant(
                        '.',
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81)],
                    lineno=8,
                    col_offset=23,
                    end_lineno=8,
                    end_col_offset=81)],
                [],
                lineno=8,
                col_offset=8,
                end_lineno=8,
                end_col_offset=82),
              lineno=8,
              col_offset=8,
              end_lineno=8,
              end_col_offset=82)],
          [
            Expr(
              Call(
                Attribute(
                  Name(
                    'phrases',
                    Load(),
                    lineno=10,
                    col_offset=8,
                    end_lineno=10,
                    end_col_offset=15),
                  'append',
                  Load(),
                  lineno=10,
                  col_offset=8,
                  end_lineno=10,
                  end_col_offset=22),
                [
                  JoinedStr(
                    [
                      Constant(
                        'And all for the want of a ',
                        lineno=10,
                        col_offset=23,
                        end_lineno=10,
                        end_col_offset=69),
                      FormattedValue(
                        Subscript(
                          Name(
                            'rhyme_items',
                            Load(),
                            lineno=10,
                            col_offset=52,
                            end_lineno=10,
                            end_col_offset=63),
                          Constant(
                            0,
                            lineno=10,
                            col_offset=64,
                            end_lineno=10,
                            end_col_offset=65),
                          Load(),
                          lineno=10,
                          col_offset=52,
                          end_lineno=10,
                          end_col_offset=66),
                        -1,
                        lineno=10,
                        col_offset=23,
                        end_lineno=10,
                        end_col_offset=69),
                      Constant(
                        '.',
                        lineno=10,
                        col_offset=23,
                        end_lineno=10,
                        end_col_offset=69)],
                    lineno=10,
                    col_offset=23,
                    end_lineno=10,
                    end_col_offset=69)],
                [],
                lineno=10,
                col_offset=8,
                end_lineno=10,
                end_col_offset=70),
              lineno=10,
              col_offset=8,
              end_lineno=10,
              end_col_offset=70)],
          lineno=7,
          col_offset=4,
          end_lineno=10,
          end_col_offset=70),
        Return(
          Name(
            'phrases',
            Load(),
            lineno=11,
            col_offset=11,
            end_lineno=11,
            end_col_offset=18),
          lineno=11,
          col_offset=4,
          end_lineno=11,
          end_col_offset=18)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=11,
      end_col_offset=18)],
  [])

# BEGIN TREE AFTER
Module(
  [
    FunctionDef(
      'placeholder_0',
      arguments(
        [],
        [],
        arg(
          'placeholder_1',
          lineno=1,
          col_offset=13,
          end_lineno=1,
          end_col_offset=24),
        [
          arg(
            'placeholder_2',
            lineno=1,
            col_offset=26,
            end_lineno=1,
            end_col_offset=35)],
        [
          None],
        defaults=[]),
      [
        If(
          UnaryOp(
            Not(),
            Name(
              'placeholder_1',
              Load(),
              lineno=3,
              col_offset=11,
              end_lineno=3,
              end_col_offset=22),
            lineno=3,
            col_offset=7,
            end_lineno=3,
            end_col_offset=22),
          [
            Return(
              List(
                [],
                Load(),
                lineno=4,
                col_offset=15,
                end_lineno=4,
                end_col_offset=17),
              lineno=4,
              col_offset=8,
              end_lineno=4,
              end_col_offset=17)],
          [],
          lineno=3,
          col_offset=4,
          end_lineno=4,
          end_col_offset=17),
        Assign(
          [
            Name(
              'placeholder_3',
              Store(),
              lineno=5,
              col_offset=4,
              end_lineno=5,
              end_col_offset=11)],
          ListComp(
            JoinedStr(
              [
                Constant(
                  'For want of a ',
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69),
                FormattedValue(
                  Name(
                    'placeholder_4',
                    Load(),
                    lineno=5,
                    col_offset=32,
                    end_lineno=5,
                    end_col_offset=41),
                  -1,
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69),
                Constant(
                  ' the ',
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69),
                FormattedValue(
                  Name(
                    'placeholder_5',
                    Load(),
                    lineno=5,
                    col_offset=48,
                    end_lineno=5,
                    end_col_offset=57),
                  -1,
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69),
                Constant(
                  ' was lost.',
                  lineno=5,
                  col_offset=15,
                  end_lineno=5,
                  end_col_offset=69)],
              lineno=5,
              col_offset=15,
              end_lineno=5,
              end_col_offset=69),
            [
              comprehension(
                Tuple(
                  [
                    Name(
                      'placeholder_4',
                      Store(),
                      lineno=6,
                      col_offset=19,
                      end_lineno=6,
                      end_col_offset=28),
                    Name(
                      'placeholder_5',
                      Store(),
                      lineno=6,
                      col_offset=30,
                      end_lineno=6,
                      end_col_offset=39)],
                  Store(),
                  lineno=6,
                  col_offset=19,
                  end_lineno=6,
                  end_col_offset=39),
                Call(
                  Name(
                    'zip',
                    Load(),
                    lineno=6,
                    col_offset=43,
                    end_lineno=6,
                    end_col_offset=46),
                  [
                    Name(
                      'placeholder_1',
                      Load(),
                      lineno=6,
                      col_offset=47,
                      end_lineno=6,
                      end_col_offset=58),
                    Subscript(
                      Name(
                        'placeholder_1',
                        Load(),
                        lineno=6,
                        col_offset=60,
                        end_lineno=6,
                        end_col_offset=71),
                      Slice(
                        Constant(
                          1,
                          lineno=6,
                          col_offset=72,
                          end_lineno=6,
                          end_col_offset=73),
                        lineno=6,
                        col_offset=72,
                        end_lineno=6,
                        end_col_offset=74),
                      Load(),
                      lineno=6,
                      col_offset=60,
                      end_lineno=6,
                      end_col_offset=75)],
                  [],
                  lineno=6,
                  col_offset=43,
                  end_lineno=6,
                  end_col_offset=76),
                [],
                0)],
            lineno=5,
            col_offset=14,
            end_lineno=6,
            end_col_offset=77),
          lineno=5,
          col_offset=4,
          end_lineno=6,
          end_col_offset=77),
        If(
          Name(
            'placeholder_2',
            Load(),
            lineno=7,
            col_offset=7,
            end_lineno=7,
            end_col_offset=16),
          [
            Expr(
              Call(
                Attribute(
                  Name(
                    'placeholder_3',
                    Load(),
                    lineno=8,
                    col_offset=8,
                    end_lineno=8,
                    end_col_offset=15),
                  'append',
                  Load(),
                  lineno=8,
                  col_offset=8,
                  end_lineno=8,
                  end_col_offset=22),
                [
                  JoinedStr(
                    [
                      Constant(
                        'And all for the want of a ',
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81),
                      FormattedValue(
                        Name(
                          'placeholder_2',
                          Load(),
                          lineno=8,
                          col_offset=52,
                          end_lineno=8,
                          end_col_offset=61),
                        -1,
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81),
                      Constant(
                        ' ',
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81),
                      FormattedValue(
                        Subscript(
                          Name(
                            'placeholder_1',
                            Load(),
                            lineno=8,
                            col_offset=64,
                            end_lineno=8,
                            end_col_offset=75),
                          Constant(
                            0,
                            lineno=8,
                            col_offset=76,
                            end_lineno=8,
                            end_col_offset=77),
                          Load(),
                          lineno=8,
                          col_offset=64,
                          end_lineno=8,
                          end_col_offset=78),
                        -1,
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81),
                      Constant(
                        '.',
                        lineno=8,
                        col_offset=23,
                        end_lineno=8,
                        end_col_offset=81)],
                    lineno=8,
                    col_offset=23,
                    end_lineno=8,
                    end_col_offset=81)],
                [],
                lineno=8,
                col_offset=8,
                end_lineno=8,
                end_col_offset=82),
              lineno=8,
              col_offset=8,
              end_lineno=8,
              end_col_offset=82)],
          [
            Expr(
              Call(
                Attribute(
                  Name(
                    'placeholder_3',
                    Load(),
                    lineno=10,
                    col_offset=8,
                    end_lineno=10,
                    end_col_offset=15),
                  'append',
                  Load(),
                  lineno=10,
                  col_offset=8,
                  end_lineno=10,
                  end_col_offset=22),
                [
                  JoinedStr(
                    [
                      Constant(
                        'And all for the want of a ',
                        lineno=10,
                        col_offset=23,
                        end_lineno=10,
                        end_col_offset=69),
                      FormattedValue(
                        Subscript(
                          Name(
                            'placeholder_1',
                            Load(),
                            lineno=10,
                            col_offset=52,
                            end_lineno=10,
                            end_col_offset=63),
                          Constant(
                            0,
                            lineno=10,
                            col_offset=64,
                            end_lineno=10,
                            end_col_offset=65),
                          Load(),
                          lineno=10,
                          col_offset=52,
                          end_lineno=10,
                          end_col_offset=66),
                        -1,
                        lineno=10,
                        col_offset=23,
                        end_lineno=10,
                        end_col_offset=69),
                      Constant(
                        '.',
                        lineno=10,
                        col_offset=23,
                        end_lineno=10,
                        end_col_offset=69)],
                    lineno=10,
                    col_offset=23,
                    end_lineno=10,
                    end_col_offset=69)],
                [],
                lineno=10,
                col_offset=8,
                end_lineno=10,
                end_col_offset=70),
              lineno=10,
              col_offset=8,
              end_lineno=10,
              end_col_offset=70)],
          lineno=7,
          col_offset=4,
          end_lineno=10,
          end_col_offset=70),
        Return(
          Name(
            'placeholder_3',
            Load(),
            lineno=11,
            col_offset=11,
            end_lineno=11,
            end_col_offset=18),
          lineno=11,
          col_offset=4,
          end_lineno=11,
          end_col_offset=18)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=11,
      end_col_offset=18)],
  [])

# BEGIN NORMALIZED CODE AFTER
def placeholder_0(*placeholder_1, placeholder_2):
    if not placeholder_1:
        return []
    placeholder_3 = [
        f"For want of a {placeholder_4} the {placeholder_5} was lost."
        for placeholder_4, placeholder_5 in zip(placeholder_1, placeholder_1[1:])
    ]
    if placeholder_2:
        placeholder_3.append(
            f"And all for the want of a {placeholder_2} {placeholder_1[0]}."
        )
    else:
        placeholder_3.append(f"And all for the want of a {placeholder_1[0]}.")
    return placeholder_3
